src/modules/rppt_tensor_color_augmentations.cpp

```diff
@@ -605,6 +605,67 @@ RppStatus rppt_contrast_host(RppPtr_t srcPtr,
     return RPP_SUCCESS;
 }
 
+/******************** lut ********************/
+
+RppStatus rppt_lut_host(RppPtr_t srcPtr,
+                        RpptDescPtr srcDescPtr,
+                        RppPtr_t dstPtr,
+                        RpptDescPtr dstDescPtr,
+                        RppPtr_t lutPtr,
+                        RpptROIPtr roiTensorPtrSrc,
+                        RpptRoiType roiType,
+                        rppHandle_t rppHandle)
+{
+    RppLayoutParams layoutParams = get_layout_params(srcDescPtr->layout, srcDescPtr->c);
+
+    if ((srcDescPtr->dataType == RpptDataType::U8) && (dstDescPtr->dataType == RpptDataType::U8))
+    {
+        lut_u8_u8_host_tensor(static_cast<Rpp8u*>(srcPtr) + srcDescPtr->offsetInBytes,
+                              srcDescPtr,
+                              static_cast<Rpp8u*>(dstPtr) + dstDescPtr->offsetInBytes,
+                              dstDescPtr,
+                              static_cast<Rpp8u*>(lutPtr),
+                              roiTensorPtrSrc,
+                              roiType,
+                              layoutParams);
+    }
+    if ((srcDescPtr->dataType == RpptDataType::U8) && (dstDescPtr->dataType == RpptDataType::F16))
+    {
+        lut_u8_f16_host_tensor(static_cast<Rpp8u*>(srcPtr) + srcDescPtr->offsetInBytes,
+                               srcDescPtr,
+                               static_cast<Rpp16f*>(dstPtr) + dstDescPtr->offsetInBytes,
+                               dstDescPtr,
+                               static_cast<Rpp16f*>(lutPtr),
+                               roiTensorPtrSrc,
+                               roiType,
+                               layoutParams);
+    }
+    if ((srcDescPtr->dataType == RpptDataType::U8) && (dstDescPtr->dataType == RpptDataType::F32))
+    {
+        lut_u8_f32_host_tensor(static_cast<Rpp8u*>(srcPtr) + srcDescPtr->offsetInBytes,
+                               srcDescPtr,
+                               static_cast<Rpp32f*>(dstPtr) + dstDescPtr->offsetInBytes,
+                               dstDescPtr,
+                               static_cast<Rpp32f*>(lutPtr),
+                               roiTensorPtrSrc,
+                               roiType,
+                               layoutParams);
+    }
+    else if ((srcDescPtr->dataType == RpptDataType::I8) && (dstDescPtr->dataType == RpptDataType::I8))
+    {
+        lut_i8_i8_host_tensor(static_cast<Rpp8s*>(srcPtr) + srcDescPtr->offsetInBytes,
+                              srcDescPtr,
+                              static_cast<Rpp8s*>(dstPtr) + dstDescPtr->offsetInBytes,
+                              dstDescPtr,
+                              static_cast<Rpp8s*>(lutPtr),
+                              roiTensorPtrSrc,
+                              roiType,
+                              layoutParams);
+    }
```

Comment:
what about other datatypes? Please return failure with error message for unsupported datatypes

Reply:
src as F16/F32 is not supported since it won't make sense, and we'll add the error message.

Reply:
error code added